File: chapter-4/Car.scala
//In Java we explicity defined the field and method for the property year and wrote an explicit constructor. Scala, the parameter took care of defining that field and writing the accessor method. Scala accomplishes the same thing wih more concise code.
class Car(val year: Int) {
  private var milesDriven: Int = 0

  def miles() = milesDriven

  def drive(distance: Int) {
    milesDriven += Math.abs(distance)
  }
}

val car = new Car(2009)
println("Car made in year " + car.year)
println("Miles driven " + car.miles)
println("Drive for 10 miles")
car.drive(10)
println("Miles driven " + car.miles)

File: chapter-4/Car.java
//In this example of a Java class, it follows the the bean convention of exposing its properties. It has two properties, and corresponding getter methods. It also has a method to manipulate the miles property. It also has a constructor which intitializes the final field year.
public class Car {
    private final int year;
    private int miles;

    public Car(int yearOfMake) {
        year = yearOfMake;
    }

    public int getYear() {
        return year;
    }

    public int getMiles() {
        return miles;
    }

    public void drive(int distance) {
        miles += Math.abs(distance);
    }
}
